solr/solrj/src/java/org/apache/solr/common/SolrDocument.java

```diff
@@ -388,20 +386,47 @@ public void addChildDocuments(Collection<SolrDocument> children) {
      }
    }
 
+   @Override
+   public Map<String, Object> getChildDocumentsMap() {
+     Map<String, Object> childDocs = new HashMap<>();
+     for (Entry<String, Object> entry: _fields.entrySet()) {
+       Object value = entry.getValue();
+       if(objIsDocument(value)) {
+         childDocs.put(entry.getKey(), value);
+       }
+     }
+     return childDocs;
+   }
+
    /** Returns the list of child documents, or null if none. */
    @Override
    public List<SolrDocument> getChildDocuments() {
-     return _childDocuments;
+     List<SolrDocument> childDocs = new ArrayList<>();
+     Stream<AbstractMap.SimpleEntry<String, SolrDocument>> fields = _fields.entrySet().stream()
+         .filter(value -> value.getValue() instanceof SolrInputDocument)
```

Comment:
Or the value might be a List of SolrInputDocument, so we have to check for that too; right?

Reply:
yeah you're right, I'll add another commit to fix this

Reply:
Instead of creating an outer List<SolrDocument> childDocs = new ArrayList<>() up front, can this look like List<...> fields = ......collect(Collectors.toList()), with the size check at the end? 
Also would be nice if this API could be changed to always return non-null, and if it's empty, it's just empty.